### PR TITLE
cleanup stderr, return caught exception if any

### PR DESCRIPTION
**Pull request by tkelman**
and whitespace in test/test.jl, best viewed with https://github.com/kshyatt/julia/pull/2/files?w=1

### CODE REVIEW

**Comment by tkelman** [comment]
I can't touch this branch any more, I got magic one-time "allow maintainers to edit" permission and it would up here... I want to tweak this again I think, let's put this in a future one.

File: test/test.jl [hunk 1]
```diff
@@ -105,69 +105,69 @@ end
...
+                end
             end
         end
-    end
-    # These lines shouldn't be called
-    redirect_stdout(OLD_STDOUT)
-    error("No exception was thrown!")
+        # These lines shouldn't be called
+        redirect_stdout(OLD_STDOUT)
+        redirect_stderr(OLD_STDERR)
```
Comment (tkelman) [inline test/test.jl:167]: I think I want these inside the catch actually

**Comment by kshyatt** [comment]
Can't you have this opened against `master` since it's from your fork and you just merged?

**Comment by tkelman** [comment]
funny thing is this isn't from my fork!

**Comment by tkelman** [comment]
I guess I could open it, but this on its own isn't worth a whole PR to base yet, may as well put it in with the next related thing

**Comment by kshyatt** [comment]
> this on its own isn't worth a whole PR to base yet, may as well put it in with the next related thing

What? Come on. I open PRs with two one-line tests all the time.

**Comment by tkelman** [comment]
I've got other silly style stuff pending in a branch I'll get this in with.

**Pull request closed without merging**